### PR TITLE
Remove remaining usages of <InstructionsDialogWrapperDEPRECATED/>

### PR DESCRIPTION
**Pull request by maddiedierker**
Removes remaining usages of `<InstructionsDialogWrapperDEPRECATED/>` and uses `<InstructionsDialog/>` instead. Outside of NetSim, which was migrated in #42622, there are two cases I could find where this dialog is still used:

#### 1. expandableImages in EnhancedSafeMarkdown
Levelbuilders can include markdown like `expandable` in `long_instructions` ([example](https://github.com/code-dot-org/code-dot-org/blob/e20c78db3cbab147940cb5144b9af19b35048b2d/dashboard/config/scripts/levels/courseC_artist_loops1a_2021.level#L43)), which uses [EnhancedSafeMarkdown](https://github.com/code-dot-org/code-dot-org/blob/47be99d6cf7df2be746b592906f50c0f3860b80a/apps/src/templates/EnhancedSafeMarkdown.jsx#L79) (as it's rendered by MarkdownInstructions, which is rendered by [Instructions](https://github.com/code-dot-org/code-dot-org/blob/47be99d6cf7df2be746b592906f50c0f3860b80a/apps/src/templates/instructions/Instructions.jsx#L56)) to turn that markdown into an image element that opens this dialog when clicked (via [renderExpandableImages](https://github.com/code-dot-org/code-dot-org/blob/47be99d6cf7df2be746b592906f50c0f3860b80a/apps/src/templates/utils/expandableImages.js#L18)).

**Example with screenshots:**
[Level courseC_artist_loops1a_2021 definition](https://github.com/code-dot-org/code-dot-org/blob/e20c78db3cbab147940cb5144b9af19b35048b2d/dashboard/config/scripts/levels/courseC_artist_loops1a_2021.level#L43) / https://studio.code.org/s/coursec-2021/lessons/10/levels/1

Long instructions with embedded expandable images:
Screen Shot 2021-10-07 at 2 32 55 PM

| Dialog before | Dialog after |
| -------------- | --------------- |
| Screen Shot 2021-10-07 at 2 33 31 PM| Screen Shot 2021-10-07 at 2 36 43 PM|

You'll notice I didn't re-implement the avatar in the dialog 😄 that is purposeful - this dialog is basically just for viewing a tiny image in a larger context, and the avatar doesn't seem contextually necessary, so I've left it off. It also caused UI weirdness on levels that don't have an avatar -- there was a blank space and weird margins around the dialog title, which we don't want.

#### 2. ani_gif_url property on a level
Levels can have an `ani_gif_url` ([example](https://github.com/code-dot-org/code-dot-org/blob/a5368c8cf6a07c6f2556d39a668aedd912ab0a74/dashboard/config/scripts/levels/allthethings-anigif-instructions.level#L14)) that is rendered as an [AniGifPreview](https://github.com/code-dot-org/code-dot-org/blob/47be99d6cf7df2be746b592906f50c0f3860b80a/apps/src/templates/instructions/AniGifPreview.jsx) in [Instructions](https://github.com/code-dot-org/code-dot-org/blob/47be99d6cf7df2be746b592906f50c0f3860b80a/apps/src/templates/instructions/Instructions.jsx#L95). Similar to above, when this image/GIF is clicked, this dialog will open.

**Example with screenshots:**
[Level allthethings-anigif-instructions definition](https://github.com/code-dot-org/code-dot-org/blob/a5368c8cf6a07c6f2556d39a668aedd912ab0a74/dashboard/config/scripts/levels/allthethings-anigif-instructions.level#L14) / https://studio.code.org/s/allthethings/lessons/2/levels/7

Instructions with AniGifPreview:
Screen Shot 2021-10-07 at 2 44 36 PM

| Dialog before | Dialog after |
| -------------- | --------------- |
|Screen Shot 2021-10-07 at 2 45 21 PM|Screen Shot 2021-10-07 at 2 46 48 PM|

(Screenshots above don't match because they're captured at different times during the GIF.)

## Links

- [STAR-1737](https://codedotorg.atlassian.net/browse/STAR-1737)

## Testing story

Updates existing tests.

## Follow-up work

- Remove [autoClose](https://github.com/code-dot-org/code-dot-org/blob/a5368c8cf6a07c6f2556d39a668aedd912ab0a74/apps/src/redux/instructionsDialog.js#L6) and [hintsOnly](https://github.com/code-dot-org/code-dot-org/blob/a5368c8cf6a07c6f2556d39a668aedd912ab0a74/apps/src/redux/instructionsDialog.js#L8) state from instructionsDialog redux. Both were always set to `false` (even before this change), so this is dead code we can remove.
- Deprecate [NonMarkdownInstructions](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/instructions/NonMarkdownInstructions.jsx) and add logging to make sure it isn't used anywhere.
- Figure out what to do with `instructions2`, which is only used in Star Wars levels ([example](https://github.com/code-dot-org/code-dot-org/blob/a5368c8cf6a07c6f2556d39a668aedd912ab0a74/apps/src/studio/levels.js#L2705)). This is a special case for Star Wars levels, so we have to keep it, but I'd like to isolate it from the rest of our instructions since it isn't used elsewhere and shouldn't be used again. (Maybe this is where NonMarkdownInstructions is still used? I'll look into that.)
- Break [instructionsComponentsTest.js](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/test/unit/instructionsComponentsTest.js) into multiple files. Why are we testing multiple components that have separate files in a single test file?

### CODE REVIEW

File: apps/src/templates/EnhancedSafeMarkdown.jsx [hunk 1]
```diff
@@ -42,20 +42,6 @@ export const ExpandableImagesWrapper = connect(
   dispatch => ({
     showImageDialog(imgUrl) {
       dispatch(
-        /*
-         * TODO: Using the 'DialogInstructions' component to display the
```
Comment (maddiedierker) [inline apps/src/templates/EnhancedSafeMarkdown.jsx:46]: DialogInstructions has been deleted, and `shortInstructions` are optional in InstructionsDialog, so this comment is no longer necessary

File: apps/src/templates/instructions/Instructions.jsx [hunk 2]
```diff
@@ -53,11 +52,8 @@ class Instructions extends React.Component {
         />
       );
     } else {
-      // Note: In this case props.shortInstructions might be undefined, but we
-      // still want to render NonMarkdownInstructions to get the puzzle title
```
Comment (maddiedierker) [inline apps/src/templates/instructions/Instructions.jsx:57]: i couldn't find any case where we want to render the puzzle title, so removing this comment and the `puzzleTitle` prop in NonMarkdownInstructions

File: apps/src/templates/instructions/NonMarkdownInstructions.jsx [hunk 3]
```diff
@@ -12,13 +12,10 @@ var styles = {
 /**
  * Non-markdown version of our instructions, displayed in a dialog when our top
  * pane instructions are not enabled.
- *
- * Currently only used by NetSim levels
```
Comment (maddiedierker) [inline apps/src/templates/instructions/NonMarkdownInstructions.jsx:16]: this wasn't true even before this refactor -- i couldn't find any NetSim levels that use these instructions. i'm going to follow up with deprecating this component and adding some logging here to see if this component is ever used

**Comment by epeach** [comment]
Love to see it! 
Screenshot from 2021-10-08 10-51-04

**Comment by epeach** [comment]
Weighing in on removing the avatar - I think that's a good call.

**Review by epeach — APPROVED** [review]
So exciting to see this level of clean-up! My one thought is that my brain struggled with "ExampleImage" name because for a second I thought it was a placeholder for tests. But that might just be my brain and how I use the word example, so no requested changes on that.

**Pull request merged**